eng/CodeAnalysis.ruleset

```diff
@@ -106,7 +106,7 @@
       <Rule Id="CA2200" Action="Warning" />          <!-- Rethrow to preserve stack details. -->
       <Rule Id="CA2201" Action="None" />             <!-- Do not raise reserved exception types -->
       <Rule Id="CA2207" Action="Warning" />          <!-- Initialize value type static fields inline -->
-      <Rule Id="CA2208" Action="None" />             <!-- Instantiate argument exceptions correctly -->
+      <Rule Id="CA2208" Action="Info" />             <!-- Instantiate argument exceptions correctly -->
```

Comment:
I would have thought we wanted Warnings here (with everything fixed or suppressed by this PR)

Reply:
We have discussed to upgrade severity level from hidden to info, so I thought I should make it same here, but yes maybe we want to make it warning to get full benefit

Reply:
Talked with @bartonjs we will make it warning after fixing/suppressing all warnings. For now, I will leave this PR on hold until applying his suggestions on improving analyzer:
- maybe we only want the analyzer to report for public/protected/protected ( do not warn for internal private methods).
- We could include the generic method parameters (and possibly the generic type parameters for constructors) as legal targets.

Then we might have much less warning need to fix/suppress, and i might update this PR with all repo fix (coreclr+libs+mono)